packages/babel-plugin-transform-react-jsx/src/transform-experimental.js

```diff
@@ -0,0 +1,815 @@
+          key = convertAttribute(attr).value;
+        } else if (attr.name.name === "__source") {
+          source = convertAttribute(attr).value;
+        } else if (attr.name.name === "__self") {
+          self = convertAttribute(attr).value;
+        } else {
+          attribs.push(attr);
+        }
+      } else {
+        attribs.push(attr);
+      }
+    }
+
+    if (attribs.length || path.node.children.length) {
+      attribs = buildJSXOpeningElementAttributes(
+        attribs,
+        file,
+        path.node.children,
+      );
+    } else {
+      // attributes should never be null
+      attribs = t.objectExpression([]);
+    }
+
+    args.push(attribs);
+
+    if (!file.opts.development) {
+      if (key !== undefined) {
+        args.push(key);
+      }
+    } else {
+      // isStaticChildren, __source, and __self are only used in development
+      args.push(
+        key === undefined ? t.identifier("undefined") : key,
+        t.booleanLiteral(path.node.children.length > 1),
+        source === undefined ? t.identifier("undefined") : source,
+        self === undefined ? t.identifier("undefined") : self,
+      );
+    }
+
+    if (opts.post) {
+      opts.post(state, file);
+    }
+
+    return (
+      state.call ||
+      t.callExpression(
+        path.node.children.length > 1 ? state.staticCallee : state.callee,
+        args,
+      )
+    );
+  }
+
+  function isChildrenProp(prop) {
+    return (
+      t.isJSXAttribute(prop) &&
+      t.isJSXIdentifier(prop.name) &&
+      prop.name.name === "children"
+    );
+  }
+
+  // Builds props for React.jsx. This function adds children into the props
+  // and ensures that props is always an object
+  function buildJSXOpeningElementAttributes(attribs, file, children) {
+    let _props = [];
+    const objs = [];
+
+    // In order to avoid having duplicate "children" keys, we avoid
+    // pushing the "children" prop if we have actual children. However,
+    // the children prop may have side effects, so to be certain
+    // these side effects are evaluated, we add them to the following prop
+    // as a sequence expression to preserve order. So:
+    // <div children={x++} foo={y}>{child}</div> becomes
+    // React.jsx('div', {foo: (x++, y), children: child});
+    // duplicateChildren contains the extra children prop values
+    let duplicateChildren = [];
+
+    const hasChildren = children && children.length > 0;
+
+    const { useSpread = false } = file.opts;
+    if (typeof useSpread !== "boolean") {
+      throw new Error(
+        "transform-react-jsx currently only accepts a boolean option for " +
+          "useSpread (defaults to false)",
+      );
+    }
+
+    const useBuiltIns = file.opts.useBuiltIns || false;
+    if (typeof useBuiltIns !== "boolean") {
+      throw new Error(
+        "transform-react-jsx currently only accepts a boolean option for " +
+          "useBuiltIns (defaults to false)",
+      );
+    }
+
+    if (useSpread && useBuiltIns) {
```

Comment:
We are going to remove these options in Babel 8, and default to `...`. We can avoid adding them to this new transform.

Reply:
just for reference, https://github.com/babel/babel/pull/11141 via [jsx label](https://github.com/babel/babel/pulls?q=is%3Apr+is%3Aopen+label%3A%22area%3A+jsx%22)

Reply:
Sounds good! I'll delete them for now and default to `...` in a separate PR?

Reply:
Would it be possible to default to ... already in this PR?